### PR TITLE
cleanup work in org.terasology.math

### PR DESCRIPTION
**Pull request by asiekierka**
### Contains

Cleanup work in org.terasology.math. Primarily marks fields which are effectively final as "final", but also performs more advanced cleanup/optimizations in Rotation.

### How to test

If the game turns on, it's probably fine.

### Outstanding before merging

- [ ] Should Transform.scale not also be final? Would changing it to final break any existing module?

### Outstanding after merging

- [x] ~~IntegerRange should be replaced with Guava's Ranges as they are much more flexible, but this is best fit for a separate PR~~ Nevermind, it would be better to improve IntegerRange.

### CODE REVIEW

**Comment by GooeyHub** [comment]
Hooray Jenkins reported success with all tests good!

File: engine/src/main/java/org/terasology/math/AABB.java [hunk 1]
```diff
@@ -452,11 +452,11 @@ public int hashCode() {
     }
 
     public boolean intersectRectangle(Vector3f from, Vector3f direction) {
-        Vector3f dirfrac = new Vector3f();
-
-        dirfrac.x = 1.0f / direction.x;
-        dirfrac.y = 1.0f / direction.y;
-        dirfrac.z = 1.0f / direction.z;
+        Vector3f dirfrac = new Vector3f(
```
Comment (vampcat) [inline engine/src/main/java/org/terasology/math/AABB.java]: Let's rename dirfrac to dirFrac, shall we?

(i realize you did not introduce this variable in the first place, but it never hurts to sprinkle some code-quality-dust over areas that you visit)

**Comment by GooeyHub** [comment]
Hooray Jenkins reported success with all tests good!

**Comment by asiekierka** [comment]
Note: The "outstanding" question is probably best answered by someone more experienced with the project.

**Comment by Cervator** [comment]
Found time to test this and the game did indeed still turn on. Merged :-)

**Pull request merged**